File: src/1769. Minimum Number of Operations to Move All Balls to Each Box/1769.go
package main

import "fmt"

// func minOperations(boxes string) []int {
// 	ans := make([]int, len(boxes))
// 	for i := range boxes {
// 		for j, c := range boxes {
// 			if c == '1' {
// 				ans[i] += abs(j - i)
// 			}
// 		}
// 	}
// 	return ans
// }

func minOperations(boxes string) []int {
	left := int(boxes[0] - '0')
	right := 0
	operations := 0
	n := len(boxes)
	for i := 1; i < n; i++ {
		if boxes[i] == '1' {
			right++
			operations += i
		}
	}
	ans := make([]int, n)
	ans[0] = operations
	for i := 1; i < n; i++ {
		operations += left - right
		if boxes[i] == '1' {
			left++
			right--
		}
		ans[i] = operations
	}
	return ans
}

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}
func main() {
	var boxes string = "110"
	fmt.Println(minOperations(boxes))
}

File: src/1769. Minimum Number of Operations to Move All Balls to Each Box/1769.py
from typing import List

class Solution:
    def minOperations(self, boxes: str) -> List[int]:
        # 1
        # left, right, operations = int(boxes[0]), 0, 0
        # for i in range(1, len(boxes)):
        #     if boxes[i] == '1':
        #         right += 1
        #         operations += i
        # res = [operations]
        # for i in range(1, len(boxes)):
        #     operations += left - right
        #     if boxes[i] == '1':
        #         left += 1
        #         right -= 1
        #     res.append(operations)
        # return res 
        # 2 
        res = []
        for i in range(len(boxes)):
            s = 0
            for j, c in enumerate(boxes):
                if c == '1':
                   s += abs(j-i)
            res.append(s)
        return res
                   
            

if __name__ == "__main__":
    boxes = "110"
    print(Solution().minOperations(boxes))
